app/src/main/java/cm/aptoide/pt/notification/ComingSoonNotificationWorker.kt

```diff
@@ -0,0 +1,78 @@
+package cm.aptoide.pt.notification
+
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import androidx.work.WorkManager
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import cm.aptoide.pt.AptoideApplication
+import cm.aptoide.pt.DeepLinkIntentReceiver
+import cm.aptoide.pt.R
+import cm.aptoide.pt.app.aptoideinstall.ComingSoonApp
+import cm.aptoide.pt.logger.Logger
+import cm.aptoide.pt.networking.image.ImageLoader
+import cm.aptoide.pt.view.app.AppCenter
+import cm.aptoide.pt.view.app.DetailedAppRequestResult
+
+class ComingSoonNotificationWorker(private val context: Context,
+                                   private val workerParameters: WorkerParameters,
+                                   private val appCenter: AppCenter) :
+    Worker(context, workerParameters) {
+
+  override fun doWork(): Result {
+    val packageName = inputData.getString(ComingSoonNotificationManager.PACKAGE_NAME)
+    appCenter.loadDetailedApp(packageName, "catappult")
+        .doOnSuccess { detailedAppResult: DetailedAppRequestResult? ->
+          if (detailedAppResult != null && detailedAppResult.detailedApp != null) {
+            cancelComingSoonVerification(packageName)
+            handleAppArrived(ComingSoonApp(
+                detailedAppResult.detailedApp.name,
+                detailedAppResult.detailedApp.icon, detailedAppResult.detailedApp.md5,
+                detailedAppResult.detailedApp.store.name, detailedAppResult.detailedApp.packageName
+            ))
+          }
+        }.toBlocking().value()
```

Comment:
We probably want to avoid continuing to propagate this practice in background operations of retrieving data from an observable datasource with toBlocking(). 
Here we can work this around using the androidx.work:work-rxjava2 lib by the sdk team that gives us an observable interface for doWork(). Just need to extend RxWorker instead of Worker. That way we should be able to avoid the toBlocking().value() and dealing with errors if necessary (probably here we just want logs).